django/thunderstore/account/forms.py

```diff
@@ -0,0 +1,125 @@
+import ulid2
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from rest_framework.authtoken.models import Token
+
+from thunderstore.account.models import ServiceAccount
+from thunderstore.repository.models import UploaderIdentity
+
+
+def create_service_account_username(id_: str) -> str:
+    return f"{id_}.sa@thunderstore.io"
+
+
+class CreateServiceAccountForm(forms.Form):
+    nickname = forms.CharField(max_length=32)
+
+    def __init__(self, user: User, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.fields["identity"] = forms.ModelChoiceField(
+            queryset=UploaderIdentity.objects.filter(members__user=user),
+        )
+
+    def clean_identity(self) -> UploaderIdentity:
+        identity = self.cleaned_data["identity"]
+        if not identity.can_create_service_account(self.user):
+            raise ValidationError("Must be identity owner to create a service account")
```

Comment:
We should probably derive the reason for a failure from within the `can_create_service_account` function itself, e.g. by adding an optional parameter for `raise=True`, which causes the function to raise a validation error rather than just return a boolean.